tests/test_cache/test_cache.py

```diff
@@ -0,0 +1,139 @@
+
+    def test_returns_allow_none_on_failure(self):
+        '''Returns a AllowNone object on exception.'''
+        self.assertIsInstance(
+            self.cache.get('http://does-not-resolve/'), cache.AllowNone)
+
+    def test_uses_default_expiration_on_failure(self):
+        '''When we get AllowNone, it uses the default expiration.'''
+        with mock.patch.object(self.cache.cache_policy, 'ttl', 17):
+            with mock.patch.object(cache.time, 'time', return_value=0):
+                self.cache.get('http://does-not-resolve/')
+                self.assertEqual(
+                    self.cache.cache['http://does-not-resolve/robots.txt'].expires, 17)
+
+    def test_robots_allowed(self):
+        '''Can check for allowed.'''
+        with requests_fixtures('test_robots_allowed'):
+            self.assertFalse(
+                self.cache.allowed('http://example.com/disallowed', 'agent'))
+            self.assertTrue(
+                self.cache.allowed('http://example.com/allowed', 'agent'))
+
+    def test_caches_robots(self):
+        '''Caches robots responses.'''
+        with requests_fixtures('test_caches_robots'):
+            self.cache.get('http://example.com/')
+
+        self.assertFalse(
+            self.cache.allowed('http://example.com/disallowed', 'agent'))
+        self.assertTrue(
+            self.cache.allowed('http://example.com/allowed', 'agent'))
+
+
+class TestAgentCache(unittest.TestCase):
+    '''Tests about AgentCache.'''
+
+    def setUp(self):
+        self.cache = cache.AgentCache('agent', 10)
+
+    def test_returns_an_agent_object(self):
+        '''Returns a Robots object.'''
+        with requests_fixtures('test_returns_an_agent_object'):
+            self.assertIsInstance(
+                self.cache.get('http://example.com/blog'), cache.Agent)
+
+    def test_allows_none_on_failure(self):
+        '''Nothing is allowed on failure.'''
+        self.assertFalse(
+            self.cache.get('http://does-not-resolve/').allowed('/path'))
+
+    def test_uses_default_expiration_on_failure(self):
+        '''On fetch failure, it uses the default expiration.'''
+        with mock.patch.object(self.cache.cache_policy, 'ttl', 17):
+            with mock.patch.object(cache.time, 'time', return_value=0):
+                self.cache.get('http://does-not-resolve/')
+                self.assertEqual(
+                    self.cache.cache['http://does-not-resolve/robots.txt'].expires, 17)
+
+    def test_agent_allowed(self):
+        '''Can check for allowed.'''
+        with requests_fixtures('test_agent_allowed'):
+            self.assertFalse(
+                self.cache.allowed('http://example.com/disallowed'))
+            self.assertTrue(
+                self.cache.allowed('http://example.com/allowed'))
+
+    def test_caches_agent(self):
+        '''Caches agent responses.'''
+        with requests_fixtures('test_caches_agent'):
+            self.cache.get('http://example.com/')
```

Comment:
Ah, that was too subtle for me, but I'm fine with it. Maybe add a comment describing that is the intention.